### PR TITLE
Update L1 suite with specification 1.3.9

### PR DESCRIPTION
**Pull request by akhilbhas**
Update the L1 test suite with the specification version 1.3.9

### CODE REVIEW

**Review by srinivasgtl — CHANGES_REQUESTED** [review]
Can you please address the below issues as well?  could not raise the conversation as there is no change on that part, but we should do some changes based on the new interface file.

Remove the references to DriverTransmitCallback in the file. line 174 to 187.


Line 622 - In test_hdmicec_hal_l1_addLogicalAddress_sinkDevice_negative, calling HdmiCecAddLogicalAddress with address 0x3 should be removed as this is no more a negative condition as per the new interface file.

1. Line 625 - In test_hdmicec_hal_l1_addLogicalAddress_sinkDevice_negative, calling HdmiCecAddLogicalAddress with address 0xF should be removed as this is no more a negative condition.  Provide a value greater than 0xf
2. API test_hdmicec_hal_l1_addLogicalAddress_sinkDevice_positive should now check for the value 0x0 and 0xF as a positive test case.  Also include one logical address of each type to check like for tuner device, playback device etc.
3. Line 824: Description for the Test test_hdmicec_hal_l1_removeLogicalAddress_sinkDevice_negative - removal of logicalAddress=0xF is not a INVALID ARGUMENT. As this is the defualt logical address it needs bit more info on how to handle this case.
4. Line 828 and 829 No need to say "API only applicable for sink devices" as this test case is now called only from a sink test. 
5. Line 856 - Remove logical Address with valid logical address but not added, is still checking the error code with HDMI_CEC_IO_SUCCESS.
6. Line 1188 and 1239: "if(logicalAddress<=0 || logicalAddress>0x0B || (logicalAddress==0x0F))" may need the Interface file update.  it shall be any logical address 0 to 0xf which is set earlier.

File: src/test_l1_hdmi_cec_driver.c [hunk 1]
```diff
@@ -503,8 +503,8 @@ void test_hdmicec_hal_l1_getPhysicalAddress_negative( void )
     result = HdmiCecGetPhysicalAddress(handle, &physicalAddress);
     if (HDMI_CEC_IO_SUCCESS  != result) { UT_FAIL("HdmiCecGetPhysicalAddress failed"); }
 
-    unsigned int maxVal = (((0x04 &0xF0 ) << 20)|( (0x04 &0x0F ) << 16) |((0x04 & 0xF0) << 4)  | (0x04 & 0x0F));
-    //Max possible physical address is 4.4.4.4
+    unsigned int maxVal = (((0x0F &0xF0 ) << 20)|( (0x0F &0x0F ) << 16) |((0x0F & 0xF0) << 4)  | (0x0F & 0x0F));
```
[the remark this answers is not in the document]
Comment (akhilbhas) [reply src/test_l1_hdmi_cec_driver.c:491]: Updated the comments form above conversation.

Logic for conversion of Physical address to maxVal still under study.

**Comment by ishthiyaqahmed** [comment]
looks fine.

**Review by ishthiyaqahmed — DISMISSED** [review]

**Review by srinivasgtl — DISMISSED** [review]
approved for interface file changes.

**Review by hari22yuva — CHANGES_REQUESTED** [review]
Not sure why we removed the test cases in the module only for the deprecated API's 
if you want to remove please remove form skeletons/src/hdmi_cec_driver.c

**Comment by akhilbhas** [comment]
Deleting the API from skeleton/src/hdmi_cec_driver.c caused build error in l3 tests.
So, brought back the deleted tests, Marked them as Deprecated in note. 
The cleanup can be done during the cleanup task.

**Review by ishthiyaqahmed — APPROVED** [review]

**Review by hari22yuva — APPROVED** [review]
Approved

**Pull request merged**